### DISCUSSION TITLE
S3 bucket for media files and SRP/SOC

### DISCUSSION THREAD

**Discussion by zhibirc** [directus/directus#18746] [Feature Requests]
### Summary

I use AWS S3 as my storage adapter.

Recently, when I experimented with S3 bucket resource-based policies, I found that it requires `s3:DeleteObject` for Directus to operate properly. I was a bit surprized because didn't expect Directus is trying to delete any file from my bucket. But I constantly got an error like this:

```
AccessDenied: Access Denied
     at throwDefaultError (/app/node_modules/@aws-sdk/smithy-client/dist-cjs/default-error-handler.js:8:22)
     at /app/node_modules/@aws-sdk/smithy-client/dist-cjs/default-error-handler.js:18:39
     at de_DeleteObjectCommandError (/app/node_modules/@aws-sdk/client-s3/dist-cjs/protocols/Aws_restXml.js:3554:12)
     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
     at async /app/node_modules/@aws-sdk/middleware-serde/dist-cjs/deserializerMiddleware.js:7:24
     at async /app/node_modules/@aws-sdk/middleware-signing/dist-cjs/middleware.js:14:20
     at async /app/node_modules/@aws-sdk/middleware-retry/dist-cjs/retryMiddleware.js:27:46
     at async /app/node_modules/@aws-sdk/middleware-logger/dist-cjs/loggerMiddleware.js:7:26
     at async DriverS3.delete (file:///app/node_modules/@directus/storage-driver-s3/dist/index.js:126:9)
     at async IncomingMessage.<anonymous> (file:///app/node_modules/@directus/api/dist/services/server.js:301:25) {
   '$fault': 'client',
```

I suspected that something could be related to deletion but until I list bucket's content I didn't realize what exactly Directus is trying to delete.

I found the some health-check related files (may be it's purpose is just to be a probe -- put and delete, to report about this sub-system healthy or not) in the bucket.

So my suggestion is re-think this approach. Because this has at least one significant drawback. If I would like to deny/prevent to delete anything from the bucket I cannot do this on AWS/bucket level, which is probably the most performant and reliable way, because of the above.

If it's not possible/planned in the nearest time, may be I could disable health-check for storage?

### Motivation

Expected outcome: storage for storing files.
Health-check shouldn't require special setup of the bucket, nor to prevent to apply any reasonable setup (permissions).

Directus 10, Node 18.12

Thank you!

### Detailed Design

-

### Requirements List

-

### Drawbacks

-

### Alternatives

-

### Adoption Strategy

-

**Comment by rijkvanzanten** [comment]
Directus will also require delete access to files for the file library to work in the first place 🤔 Are you trying to disable file management completely?

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
